packages/remix-server-runtime/data.ts

```diff
@@ -79,11 +82,11 @@ export async function callRouteLoader({
   loadContext: AppLoadContext;
 }) {
   if (!loader) {
-    throw new Error(
-      `You made a ${request.method} request to ${request.url} but did not provide ` +
-        `a default component or \`loader\` for route "${routeId}", ` +
-        `so there is no way to handle the request.`
-    );
+    let pathname = new URL(request.url).pathname;
+    let msg =
+      `You made a ${request.method} request to "${pathname}" but did not provide ` +
+      `a \`loader\` for route "${routeId}", so there is no way to handle the request.`;
+    throw new ErrorResponse(405, "Method Not Allowed", new Error(msg), true);
```

Comment:
Jacob and Matt to think a bit more on 405 or 404 here

Reply:
Landed on a 400 since it seemed the most accurate: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses

> The server cannot or will not process the request due to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).